TCCTF_Structures_v3/Entities/Structures/Altar/Dragonfriend/AltarDragonfriend.as

```diff
@@ -111,79 +100,55 @@ void onTick(CBlob@ this)
 
 	if (client)
 	{
-		buy_pressed = false;	
-		sell_pressed = false;	
-	
 		const f32 power = this.get_f32("deity_power");
 	
-		const f32 stonks_volatility = this.get_f32("stonks_volatility");
 		const f32 stonks_growth = this.get_f32("stonks_growth");
 		const f32 stonks_value = this.get_f32("stonks_value");
 		
-		string volatility_text;
-		if (stonks_volatility > 0.90f) volatility_text = "extremely high";
-		else if (stonks_volatility > 0.80f) volatility_text = "very high";
-		else if (stonks_volatility > 0.60f) volatility_text = "high";
-		else if (stonks_volatility > 0.40f) volatility_text = "medium";
-		else if (stonks_volatility > 0.20f) volatility_text = "low";
-		else volatility_text = "very low";
-		
 		string text = "Altar of the Dragon\n";
 		text += "\nDragon Power: " + power;
 		text += "\nFire Resistance: " + Maths::Min(power / power_fire_immunity_max * 100.00f, 100.00f) + "%";
 		text += "\nMaximum Stonks Value: " + Maths::Ceil(stonks_base_value_max + (power / 100.00f)) + " coins";
 		text += "\nFireball Power: " + Maths::Round((1.00f + Maths::Sqrt(power * 0.00002f)) * 100.00f) + "%";
-		// text += "\n\nStonks:";
-		text += "\nVolatility: " + volatility_text;
-		// text += "\nGrowth: " + (stonks_growth >= 0 ? "+" : "-") + (Maths::Abs(s32(stonks_growth * 10000.00f) * 0.01f)) + "%";
-		// text += "\n";
-		// text += "\nSell Price: " + Maths::Ceil(stonks_value) + " coins";
-		// text += "\nBuy Price: " + Maths::Ceil(stonks_value * 0.98f) + " coins";
 		this.setInventoryName(text);
 		
 		const f32 radius = 64.00f + ((power / 100.00f) * 8.00f);
 		this.SetLightRadius(radius);
 	}
 	
-	if (isServer())
+	updateStonks(this);
```

Comment:
updateStonks(this,20*XORRandom(2)-10); //give random -10 or 10

Reply:
no need is good ingame, check it out